ucdavis_service/src/main/java/com/ucdavis/application/UCDavisService.java

```diff
@@ -135,13 +142,87 @@ public void writeToMap(List<JSONObject> records, String target) {
 			//System.out.println("TmpBase: " + tmpBase.toString());
 			dataItems.put(key, tmpBase);
 		}
-		if (data.containsKey(target)) {
-			data.get(target).putAll(dataItems);
+		if (data.containsKey(request.getTargets())) {
+			data.get(request.getTargets()).putAll(dataItems);
 		} else {
-			data.put(target, dataItems);
+			data.put(request.getTargets(), dataItems);
+		}
+		
+		fillMissingDateKeys(dataItems, request); // date sequence is not continuous
+		checkNullKeysAndValue(dataItems, true); // check the null column and value
+	}
+	
+	private void fillMissingDateKeys(Map<String, Map<String,String>> dataItems,  Request request) {
+		String start = request.getStart_date();
+		String end = request.getEnd_date();
+		
+		while (!start.equals(end)) {
+			// Initialize the dateMap
+			Map<String, String> dateMap = new HashMap<>();
+			//case 1: no find "Date" key
+			if (!dataItems.containsKey("Date")) {
+				dateMap.put(start, start);
+				dataItems.put("Date", dateMap);
+			} else {
+				// case 2: "Date" key exists already, check date is completed
+				if (!dataItems.get("Date").containsKey(start)) {
+					dateMap.put(start, start);
+					dataItems.get("Date").putAll(dateMap);
+					start = this.addOneDay(start);
+				} 
+				// case 3: "Date" key exists, date sequence completed
+				else {
+					start = this.addOneDay(start);
+					continue;
+				}
+			}
+			// iterate
+			start = this.addOneDay(start);
 		}
 	}
 	
+	private void checkNullKeysAndValue(Map<String, Map<String, String>> dataItems, boolean isDaily) {
+		// daily keys
```

Comment:
Daily check can be checked from json file, the Boolean variable should not be necessary